### PR TITLE
[TensorExpr] Add NNC lowerings for `aten::mean`, `aten::addmm`, and `aten::adaptive_avg_pool2d`.

### PR DESCRIPTION
**Pull request by ZolotukhinM**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59347 [TensorExpr] Add NNC lowerings for `aten::mean`, `aten::addmm`, and `aten::adaptive_avg_pool2d`.**
* #59346 [TensorExpr] Handle not-specified dtypes and strides.
* #59163 [JIT] ShapeProp: add missing ops from mobilenet v3.

We had external call wrappers for them, but they were not used in NNC.
This PR adds lowerings using these ext calls and fixes some bugs in
them.

Differential Revision: [D28853832](https://our.internmc.facebook.com/intern/diff/D28853832)

### CODE REVIEW

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

File: torch/csrc/jit/tensorexpr/external_functions.cpp [hunk 1]
```diff
@@ -110,7 +110,10 @@ void nnc_aten_adaptive_avg_pool2d(
   at::Tensor& r = tensors[0];
   const at::Tensor& x = tensors[1];
   int64_t H = extra_args[0];
-  int64_t W = extra_args[1];
+  int64_t W = H;
+  if (args_num > 1) {
```
Comment (Chillee) [inline torch/csrc/jit/tensorexpr/external_functions.cpp:114]: In my opinion, we should always pass in all of the args, and handle this kind of default propagation at the lowerings level.

**Review by Chillee — APPROVED** [review]
For some of these, I think I'm pretty close to being able to codegen them.

File: torch/csrc/jit/tensorexpr/external_functions.cpp (hunk 1 above)
Comment (ZolotukhinM) [reply torch/csrc/jit/tensorexpr/external_functions.cpp:114]: That sounds good to me, I will update the PR!

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

**Comment by ZolotukhinM** [comment]
@ZolotukhinM has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28853832).

**Pull request closed without merging**